comp/netflow/goflowlib/convert.go

```diff
@@ -83,63 +89,69 @@ func applyAdditionalField(flow *common.Flow, destination string, fieldValue any)
 	// Make sure FlowFieldsTypes includes the type of the following fields
 	switch destination {
 	case "direction":
-		setValue(&flow.Direction, fieldValue)
+		setInt(&flow.Direction, fieldValue)
 	case "start":
-		setValue(&flow.StartTimestamp, fieldValue)
+		setInt(&flow.StartTimestamp, fieldValue)
 	case "end":
-		setValue(&flow.EndTimestamp, fieldValue)
+		setInt(&flow.EndTimestamp, fieldValue)
 	case "bytes":
-		setValue(&flow.Bytes, fieldValue)
+		setInt(&flow.Bytes, fieldValue)
 	case "packets":
-		setValue(&flow.Packets, fieldValue)
+		setInt(&flow.Packets, fieldValue)
 	case "ether_type":
-		setValue(&flow.EtherType, fieldValue)
+		setInt(&flow.EtherType, fieldValue)
 	case "ip_protocol":
-		setValue(&flow.IPProtocol, fieldValue)
+		setInt(&flow.IPProtocol, fieldValue)
 	case "exporter.ip":
-		setValue(&flow.ExporterAddr, fieldValue)
+		setBytes(&flow.ExporterAddr, fieldValue)
 	case "source.ip":
-		setValue(&flow.SrcAddr, fieldValue)
+		setBytes(&flow.SrcAddr, fieldValue)
 	case "source.port":
 		var port uint64
-		setValue(&port, fieldValue)
+		setInt(&port, fieldValue)
 		flow.SrcPort = int32(port)
 	case "source.mac":
-		setValue(&flow.SrcMac, fieldValue)
+		setInt(&flow.SrcMac, fieldValue)
 	case "source.mask":
-		setValue(&flow.SrcMask, fieldValue)
+		setInt(&flow.SrcMask, fieldValue)
 	case "destination.ip":
-		setValue(&flow.DstAddr, fieldValue)
+		setBytes(&flow.DstAddr, fieldValue)
 	case "destination.port":
 		var port uint64
-		setValue(&port, fieldValue)
+		setInt(&port, fieldValue)
 		flow.DstPort = int32(port)
 	case "destination.mac":
-		setValue(&flow.DstMac, fieldValue)
+		setInt(&flow.DstMac, fieldValue)
 	case "destination.mask":
-		setValue(&flow.DstMask, fieldValue)
+		setInt(&flow.DstMask, fieldValue)
 	case "ingress.interface":
-		setValue(&flow.InputInterface, fieldValue)
+		setInt(&flow.InputInterface, fieldValue)
 	case "egress.interface":
-		setValue(&flow.OutputInterface, fieldValue)
+		setInt(&flow.OutputInterface, fieldValue)
 	case "tcp_flags":
-		setValue(&flow.TCPFlags, fieldValue)
+		setInt(&flow.TCPFlags, fieldValue)
 	case "next_hop.ip":
-		setValue(&flow.NextHop, fieldValue)
+		setBytes(&flow.NextHop, fieldValue)
 	case "tos":
-		setValue(&flow.Tos, fieldValue)
+		setInt(&flow.Tos, fieldValue)
 	default:
 		return false
 	}
 	return true
 }
 
-type flowFieldsTypes interface {
-	uint64 | uint32 | []byte
+func setInt[T uint64 | uint32](field *T, value any) {
+	if v, ok := value.(uint64); ok {
+		*field = T(v)
```

Comment:
I'm not sure to fully understand this logic, we won't set anything if the value is uint32?

Reply:
ah, I see, we are evaluating the type of the value here, and the type of the value values is always uint64 for integer: https://github.com/DataDog/datadog-agent/blob/c90ae4b33ef02d282eba2b63a8129dcc1701cf1a/comp/netflow/goflowlib/additionalfields/netflow.go#L28-L34

Reply:
Value is always gonna be `uint64`, since integer additional fields are always converted to `uint64` [here](https://github.com/DataDog/datadog-agent/blob/9b7db86221e5e1f157345b3955fe25a71c9a7e5e/comp/netflow/goflowlib/additionalfields/netflow.go#L20)
Here, we're converting these to `uint64` or `uint32` depending on the type of the underlying field to override. We do that by using go generics `*field = T(v)` where `T` is the type of `field`